src/yggdrasil/core.go

```diff
@@ -187,4 +191,12 @@ func (c *Core) Stop() {
 // This function is unsafe and should only be ran by the core actor.
 func (c *Core) _stop() {
 	c.log.Infoln("Stopping...")
+	if c.addPeerTimer != nil {
+		c.addPeerTimer.Stop()
+	}
+	c.link.stop()
+	for _, peer := range c.GetPeers() {
```

Comment:
This races with the `c.link.stop()`, if there are any connections that have started (either still `Dial`ing, or successfully `Dial`ed or `Accept`ed but not done running the link handler) but which haven't created a peer yet by the time we get here.

Reply:
Race should be fixed now, `c.link.stop()` will now signal links to exit before calling `tcp.stop()`, and `tcp.stop()` will block until all listeners have exited and all connections have closed (via a waitgroup). This could probably be done a little cleaner with a `util.Cancellation` but what I'd really like to do is get rid of the idle goroutines that sit around waiting for something to unblock, so we should come back to that later.